### PR TITLE
Lesson OOP

### PR DESCRIPTION
**Pull request by sergbilash**

### CODE REVIEW

File: src/oop/app/src/Scrapper.php [hunk 1]
```diff
@@ -7,3 +7,28 @@
...
+namespace src\oop\app\src;
+
+use src\oop\app\src\Models\Movie;
+
+class Scrapper
+{
+    public $transporter;
+    public $parser;
+
+    public function __construct($transporter, $parser)
```
Comment (domioedition) [inline src/oop/app/src/Scrapper.php:20]: Where type hinting?

File: src/oop/app/src/ScrapperFactory.php [hunk 2]
```diff
@@ -19,9 +21,9 @@ public function create(string $domain): Scrapper
     {
         switch ($domain) {
             case 'filmix':
-                return new Scrapper(new CurlStrategy(), new FilmixParserStrategy());
+                return new Scrapper(new CurlStrategy(), new FilmixParserStrategy(new Movie()));
             case 'kinoukr':
-                return new Scrapper(new GuzzleAdapter(), new KinoukrDomCrawlerParserAdapter());
+                return new Scrapper(new GuzzleAdapter(new Client()), new KinoukrDomCrawlerParserAdapter(new Movie()));
```
Comment (domioedition) [inline src/oop/app/src/ScrapperFactory.php]: Не надо передавать тут new Client() и new Movie()

File: src/oop/app/src/Transporters/GuzzleAdapter.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+use \GuzzleHttp\Client;
+
+class GuzzleAdapter implements TransportInterface
+{
+    private Client $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
```
Comment (domioedition) [inline src/oop/app/src/Transporters/GuzzleAdapter.php]: $this->client = new Client()

File: src/oop/app/src/Parsers/FilmixParserStrategy.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+     * @param string $siteContent
+     * @return mixed
+     */
+    public function parseContent(string $siteContent): Movie
+    {
+        preg_match('/<h1[^>]*>(.*?)<\/h1>/', $siteContent, $title);
+        preg_match('/<img.*?src=[\'"](https.*)[\'"].*?class="poster.*>/', $siteContent, $poster);
+        preg_match('/<div.*class="about".*?><div.*?>(.*?)<\/div>/', $siteContent, $description);
+
+        return $this->movie->setTitle($title[1])->setPoster($poster[1])->setDescription($description[1]);
```
Comment (domioedition) [inline src/oop/app/src/Parsers/FilmixParserStrategy.php]: так работать не будет.

**Comment by sergbilash** [comment]
"[PHP-WEB] fix issues"  - this should be read as "[PHP-OOP] fix issues"
After pushing commits to repository I can't to change commit message by the command: "git commit --amend -m "New commit message.""

**Comment by domioedition** [comment]
✔👍

**Review by domioedition — APPROVED** [review]
👍

**Pull request merged**